window.go

```diff
@@ -50,24 +50,53 @@ func (w *Window) Until() time.Duration {
 		return time.Duration(0)
 	}
 
-	now := normaliseTime(w.Now())
-	if w.xMidnight && now.Before(w.Start) {
-		now = now.Add(24 * time.Hour)
-	}
+	now := w.nowTimeAfterStart()
 
-	untilStart := w.Start.Sub(now)
-	if untilStart > 0 {
-		// Before window start.
-		return untilStart
-	}
-	if w.End.Sub(now) >= 0 {
+	if w.End.After(now) || w.End == now {
 		// During window.
 		return time.Duration(0)
 	}
 	// After window.
-	return w.Start.Add(24 * time.Hour).Sub(now)
+	return addDay(w.Start).Sub(now)
+}
+
+func addDay(t time.Time) time.Time {
+	return t.Add(24 * time.Hour)
+}
+
+// to make time calculations easier we choose a date time to represent now
```

Comment:
In Go, function comments should start with the function name. E.g. "nowTimeAfterStart calculates the blah from the foo using a pinch of bar". There are tools which rely on this for matching up documentation with the thing being documented.